### ISSUE TITLE
async version of gen_rlock/gen_wlock should not be async function

### ISSUE THREAD

**Issue by kice** [elarivie/pyReaderWriterLock#8]
I don't think they should be `async` function, since those functions do not await anything.

**Comment by elarivie** [comment]
Hello @kice 

Thanks for your observation and analyze of the code.  Code reviews are welcome.

Effectively their current implementation do not await anything because they do not need to.

But the methods comply with the parent Protocol(aka interface) of RWLockable & RWLockableD

* https://github.com/elarivie/pyReaderWriterLock/blob/5569662f3cab997a22b50522794afe0258c09850/readerwriterlock/rwlock_async.py#L99

* https://github.com/elarivie/pyReaderWriterLock/blob/5569662f3cab997a22b50522794afe0258c09850/readerwriterlock/rwlock_async.py#L112

Another implementation using a different internal strategy could await on something, for example an implementation which internally use a file on disk to store states could await stuff within the method.

Basically the fact that the methods internally await or not become implementation detail.

What is your use case, is the current implementation causing issues?
